c++/test/titerate.cpp

```diff
@@ -162,18 +162,18 @@ test_iter_group(FileAccPropList &fapl)
             DataSet dataset = file.createDataSet(name, datatype, filespace);
 
             /* Keep a copy of the dataset names */
-            lnames[i] = HDstrdup(name);
-            check_values(lnames[i], "HDstrdup returns NULL", __LINE__, __FILE__);
+            lnames[i] = strdup(name);
+            check_values(lnames[i], "strdup returns NULL", __LINE__, __FILE__);
         }
 
         /* Create a group and named datatype under root group for testing */
         Group grp(file.createGroup(GROUP1, 0));
-        lnames[NDATASETS] = HDstrdup("grp");
-        check_values(lnames[NDATASETS], "HDstrdup returns NULL", __LINE__, __FILE__);
+        lnames[NDATASETS] = strdup("grp");
```

Comment:
strdup is only POSIX right? Do we want to be including that here?

Reply:
I thought about it, but all C compilers I know of support it, so I'm letting it slide. I can switch it back if people feel strongly about it.

Reply:
Though eventually it won't matter, we'll assume a POSIX environment and have quiet work-arounds for Windows (i.e., no HDfoo - just foo). We'll only use an H5_ prefix for calls that are quirky, like basename/dirname and getopt.

Reply:
Right, I'm not sure if there are any platforms where it would be a problem anymore or not. Definitely still issues for `strndup`, but not sure on `strdup`